### PR TITLE
LF-REPL: proper dev-mode

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
We add a flag in LF-REPL to prevent usage of DAML-LF Dev and dev value/transaction version.

This PR advances #5164.

## Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-lf/tests/scenario/test.sh [hunk 1]
```diff
@@ -36,13 +36,20 @@ REPL=$(rlocation "$TEST_WORKSPACE/$1")
 DAMLC=$(rlocation "$TEST_WORKSPACE/$2")
 TESTMAIN=$(rlocation "$TEST_WORKSPACE/$3")
 DIFF="$4"
+if [ "$5" == true ] ; then
```
Comment (cocreature) [inline daml-lf/tests/scenario/test.sh:39]:
```suggestion
if [ "$5" = "true" ] ; then
```

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala [hunk 2]
```diff
@@ -562,6 +562,10 @@ object Speedy {
         compiledPackages: CompiledPackages,
         transactionSeed: crypto.Hash,
         scenario: SExpr,
+        supportedValueVersions: VersionRange[value.ValueVersion] =
+          value.ValueVersions.DefaultSupportedVersions,
```
Comment (cocreature) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala:566]: Just to confirm my understanding: The default still includes the dev versions?

File: daml-lf/repl/src/main/scala/com/digitalasset/daml/lf/repl/testing/Main.scala [hunk 3]
```diff
@@ -64,25 +64,25 @@ object Main extends App {
     System.exit(1)
   } else {
     var replArgs = args.toList
-    var allowDev = false
+    var devMode = false
     replArgs match {
-      case "--decode-lfdev" :: rest =>
+      case "--dev" :: rest =>
```
Comment (cocreature) [inline daml-lf/repl/src/main/scala/com/digitalasset/daml/lf/repl/testing/Main.scala:69]: Maybe time to switch to a proper CLI parser?

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala (hunk 2 above)
Comment (remyhaemmerle-da) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala:566]: You are right. I rename the constant to 'SupportedDevVersions'

File: daml-lf/repl/src/main/scala/com/digitalasset/daml/lf/repl/testing/Main.scala (hunk 3 above)
Comment (remyhaemmerle-da) [reply daml-lf/repl/src/main/scala/com/digitalasset/daml/lf/repl/testing/Main.scala:69]: maybe in another PR.

**Pull request merged**